packet/utils.py

```diff
@@ -63,19 +64,24 @@ def get_member_info(uid):
 
 @lru_cache(maxsize=2048)
 def is_on_floor(uid):
-    return get_freshman(uid).onfloor
-
-
+    freshman = Freshman.query.filter_by(rit_username=uid).first()
+    if freshman is not None:
+        return freshman.onfloor
+    else:
+        return False
+
+
+def packet_auth(func):
+    """
+    Decorator for easily configuring oidc
+    """
+    @auth.oidc_auth
+    @wraps(func)
+    def wrapped_function(*args, **kwargs):
+        if app.config["REALM"] == "csh":
+            if ldap_is_intromember(ldap_get_member(str(session["userinfo"].get("preferred_username", "")))):
+                return "Sorry, upperclassmen packet is not available to intro members.", 401
```

Comment:
Don't just return strings, show a flash message or an error page.

Reply:
We'll make this more user-friendly in the new UI but for now we just need it to work.